numalogic/registry/mlflow_registry.py

```diff
@@ -127,25 +136,20 @@ def load(
     ) -> Optional[ArtifactData]:
         model_key = self.construct_key(skeys, dkeys)
         try:
-            if latest:
-                model = self.handler.load_model(
-                    model_uri=f"models:/{model_key}/{ModelStage.PRODUCTION}"
-                )
-                version_info = self.client.get_latest_versions(
-                    model_key, stages=[ModelStage.PRODUCTION]
-                )[-1]
-            elif version is not None:
-                model = self.handler.load_model(model_uri=f"models:/{model_key}/{version}")
-                version_info = self.client.get_model_version(model_key, version)
-            else:
-                raise ValueError("One of 'latest' or 'version' needed in load method call")
-            _LOGGER.info("Successfully loaded model %s from Mlflow", model_key)
 
-            run_info = mlflow.get_run(version_info.run_id)
-            metadata = run_info.data.params or None
-            _LOGGER.info("Successfully loaded model metadata from Mlflow!")
+            if (latest and version) or (not latest and not version):
+                raise ValueError("One of 'latest' or 'version' needed in load method call")
 
+            elif latest:
+                version_info = self.client.get_latest_versions(model_key, stages=[self.model_stage])
+                if not version_info:
+                    raise ValueError("No Model found in %s" % self.model_stage)
```

Comment:
Raise a custom exception here; kind of how we talked about earlier

Reply:
Also at line 162, can you change the log to `Unexpected Error..`

Reply:
Cool